event-hubs.api.md

```diff
@@ -0,0 +1,287 @@
+    close(): Promise<void>;
+    static defaultConsumerGroupName: string;
+    getEventHubProperties(options?: GetEventHubPropertiesOptions): Promise<EventHubProperties>;
+    getPartitionIds(option?: GetPartitionIdsOptions): Promise<string[]>;
+    getPartitionProperties(partitionId: string, options?: GetPartitionPropertiesOptions): Promise<PartitionProperties>;
+    subscribe(consumerGroupName: string, handlers: SubscriptionEventHandlers, options?: SubscriptionOptions): Subscription;
+    subscribe(consumerGroupName: string, partitionId: string, handlers: SubscriptionEventHandlers, options?: SubscriptionOptions): Subscription;
+    subscribe(consumerGroupName: string, partitionManager: PartitionManager, handlers: SubscriptionEventHandlers, options?: SubscriptionOptions): Subscription;
+}
+
+// @public
+export class EventHubProducerClient {
+    constructor(connectionString: string, options?: EventHubClientOptions);
+    constructor(connectionString: string, eventHubName: string, options?: EventHubClientOptions);
+    constructor(host: string, eventHubName: string, credential: TokenCredential, options?: EventHubClientOptions);
+    close(): Promise<void>;
+    createBatch(options?: CreateBatchOptions): Promise<EventDataBatch>;
+    readonly eventHubName: string;
+    readonly fullyQualifiedNamespace: string;
+    getEventHubProperties(options?: GetEventHubPropertiesOptions): Promise<EventHubProperties>;
+    getPartitionIds(options?: GetPartitionIdsOptions): Promise<Array<string>>;
+    sendBatch(batch: EventDataBatch, options?: SendBatchOptions): Promise<void>;
+}
+
+// @public
+export interface EventHubProperties {
+    createdAt: Date;
+    name: string;
+    partitionIds: string[];
+}
+
+// @public
+export class EventPosition {
+    // Warning: (ae-forgotten-export) The symbol "EventPositionOptions" needs to be exported by the entry point index.d.ts
+    // 
+    // @internal
+    constructor(options?: EventPositionOptions);
+    static earliest(): EventPosition;
+    enqueuedTime?: Date | number;
+    static fromEnqueuedTime(enqueuedTime: Date | number): EventPosition;
+    static fromOffset(offset: number, isInclusive?: boolean): EventPosition;
+    static fromSequenceNumber(sequenceNumber: number, isInclusive?: boolean): EventPosition;
+    isInclusive: boolean;
+    static latest(): EventPosition;
+    offset?: number | "@latest";
+    sequenceNumber?: number;
+    }
+
+// @public
+export function extractSpanContextFromEventData(eventData: EventData): SpanContext | undefined;
+
+// @public
+export interface GetEventHubPropertiesOptions extends AbortSignalOptions, SpanOptions {
+}
+
+// @public
+export interface GetPartitionIdsOptions extends AbortSignalOptions, SpanOptions {
+}
+
+// @public
+export interface GetPartitionPropertiesOptions extends AbortSignalOptions, SpanOptions {
+}
+
+// @public
+export class InMemoryPartitionManager implements PartitionManager {
```

Comment:
This should no longer be in the package. It should either be removed or moved to samples.

Reply:
yes, Python will remove it too

Reply:
There were talks about this being useful for users to use in their test code. We can keep it out of GA, reach a consensus post GA and take a call as a team